### PR TITLE
Added the InternalPartition.isLocal method

### PR DESCRIPTION
**Pull request by pveentjer**
Instead of doing a thisAddress.equals(internalPartition.getOwnerOrNull). But under the hood the same functionality is used.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/partition/InternalPartition.java [hunk 1]
```diff
@@ -33,6 +33,18 @@
     int MAX_BACKUP_COUNT = MAX_REPLICA_COUNT - 1;
 
     /**
+     * Checks if the partition is local.
+     *
+     * A partition is local if and only if the {@link #getOwnerOrNull()} returns the same address as 'this' address of the
+     * {@link com.hazelcast.cluster.ClusterService#getThisAddress()}. If the address is null or a different address, false
+     * is returned.
+     *
+     * @return true of local, false otherwise.
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/partition/InternalPartition.java]: typo. true IF local

**Comment by jerrinot** [comment]
there is a typo, otherwise :+1:

File: hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionImpl.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import java.util.Arrays;
 
 class InternalPartitionImpl implements InternalPartition {
-
+;
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionImpl.java]: This semicolon feels lonely :)

**Comment by Donnerbart** [comment]
Nice little change, another removal of a `NodeEngine` dependency in the code :)

**Comment by pveentjer** [comment]
I'm working currently on the caller runs scheduler (renamed to progressive scheduler). And I need to know which partitions are locally available so the thiefs know which partitions they can rob if they have nothing better to do. So I ran into the lack of the isLocal method

**Comment by Donnerbart** [comment]
verify

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**